spec/models/manageiq/providers/ansible_tower/configuration_manager/refresh_parser_spec.rb

```diff
@@ -18,18 +18,26 @@
   let(:all_job_templates) { (1..2).collect { |i| AnsibleTowerClient::JobTemplate.new(mock_api, "id" => i, "name" => "template#{i}", "description" => "description#{i}", "extra_vars" => "some_json_payload") } }
 
   it "#configuration_manager_inv_to_hashes" do
+    vm = FactoryGirl.create(:vm_redhat, :uid_ems => "vmwareVm-2")
     allow_any_instance_of(AnsibleTowerClient::JobTemplate).to receive(:survey_spec).and_return('some_hash_payload')
     expect(manager.provider).to receive(:connect).and_return(connection)
 
     parser.configuration_manager_inv_to_hashes
 
     expect(parser.instance_variable_get(:@data)[:configured_systems].count).to eq(2)
     expect(parser.instance_variable_get(:@data)[:configured_systems].first).to eq(
-      :manager_ref          => "1",
+      :counterpart_id       => nil,
+      :counterpart_type     => nil,
       :hostname             => "host1",
       :inventory_root_group => {:ems_ref => "1", :name => "inventory1", :type => "ManageIQ::Providers::ConfigurationManager::InventoryRootGroup"},
+      :manager_ref          => "1",
       :type                 => "ManageIQ::Providers::AnsibleTower::ConfigurationManager::ConfiguredSystem",
-      :virtual_instance_ref => "vmwareVm-1"
+      :virtual_instance_ref => "vmwareVm-1",
```

Comment:
What if there are multiple EmsRedhat systems and each one has a vmwareVm-1?  Is uid_ems globally unique, since that's what the column is for?  If not, we shouldn't use that and should link to ems_ref, ems_id or something